### PR TITLE
Add config apply integration test

### PR DESCRIPTION
**Pull request by dhurley**
### Proposed changes

Add config apply integration tests

### Checklist

Before creating a PR, run through this checklist and mark each as complete.

- [ ] I have read the [`CONTRIBUTING`](https://github.com/nginx/agent/blob/main/docs/CONTRIBUTING.md) document
- [ ] I have run ```make install-tools``` and have attached any dependency changes to this pull request
- [ ] If applicable, I have added tests that prove my fix is effective or that my feature works
- [ ] If applicable, I have checked that any relevant tests pass after adding my changes
- [ ] If applicable, I have updated any relevant documentation ([`README.md`](https://github.com/nginx/agent/blob/main/README.md))
- [ ] If applicable, I have tested my cross-platform changes on Ubuntu 22, Redhat 8, SUSE 15 and FreeBSD 13

### CODE REVIEW

File: test/integration/grpc_management_plane_api_test.go [hunk 1]
```diff
@@ -177,16 +187,102 @@ func TestGrpc_ConfigUpload(t *testing.T) {
 	require.NoError(t, err)
 	assert.Equal(t, http.StatusOK, resp.StatusCode())
 
-	client = resty.New()
+	responses := getManagementPlaneResponses(t, 2)
+
+	assert.Equal(t, mpi.CommandResponse_COMMAND_STATUS_OK, responses[0].GetCommandResponse().GetStatus())
```
Comment (aphralG) [inline test/integration/grpc_management_plane_api_test.go:192]: Why is this being checked twice ?

File: test/integration/grpc_management_plane_api_test.go [hunk 2]
```diff
@@ -177,16 +187,102 @@ func TestGrpc_ConfigUpload(t *testing.T) {
...
+		fmt.Sprintf("/mock-management-plane-grpc/config/%s/etc/nginx/nginx.conf", nginxInstanceID),
+		0o666,
+	)
+	require.NoError(t, err)
+
+	performConfigApply(t, nginxInstanceID)
+
+	responses = getManagementPlaneResponses(t, 5)
+	t.Logf("Config apply responses: %v", responses)
+	assert.Equal(t, mpi.CommandResponse_COMMAND_STATUS_OK, responses[0].GetCommandResponse().GetStatus())
```
Comment (aphralG) [inline test/integration/grpc_management_plane_api_test.go:239]: Should we add a test to check if config apply fails but rollback isn't needed ?

File: test/integration/grpc_management_plane_api_test.go (hunk 1 above)
Comment (dhurley) [reply test/integration/grpc_management_plane_api_test.go:192]: The mock management plane is sending a config upload request on startup automatically. Thats why there are 2 responses

File: test/integration/grpc_management_plane_api_test.go (hunk 2 above)
Comment (dhurley) [reply test/integration/grpc_management_plane_api_test.go:239]: let be see if I can add a test for that

Comment (aphralG) [reply test/integration/grpc_management_plane_api_test.go:239]: Happens if file isn't in allowed directories might be the easiest way to try test it

Comment (dhurley) [reply test/integration/grpc_management_plane_api_test.go:259]: Added a new test for that

**Review by craigell — APPROVED** [review]

**Review by aphralG — APPROVED** [review]

**Pull request merged**